nile-backend/src/main/java/com/tp/Nile/repositories/ProductRepository.java

```diff
@@ -17,9 +17,9 @@
     @Query("select p from Product p where p.category_id=:category")
```

Comment:
Duplicate methods. I would refrain from using a custom query when there are predefined ways to create these methods using JPA and Hibernate.